### PR TITLE
chore: add flake8 pre-commit check for harness

### PR DESCRIPTION
**Pull request by hamidzr**
## Commentary (optional)

we have 9 different copies of flake8 configs. do we want the same style choices everywhere?
./cli
./common
./deploy
./e2e_tests
./examples/tests
./harness
./model_hub
./schemas
./tools

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -74,6 +74,13 @@ repos:
     hooks:
       - id: black
         exclude: ^(harness/determined/common/api/bindings.py)
+  - repo: https://github.com/pycqa/flake8
+    # Note: this should be kept in sync with the version in `requirements.txt`.
```
Comment (loksonarius) [inline .pre-commit-config.yaml:78]: Would setting it for the global requirements file not be enough? Presumably if flake is in PATH, it's the one from the requirements file.
I think we're more likely to forget to update both versions than we are to be bitten in the butt by a version mismatch.

**Review by loksonarius — APPROVED** [review]
Generally approve of changes -- one comment as a point of discussion.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (hamidzr) [reply .pre-commit-config.yaml:78]: IIRC pre-commit installs and manages its own virtual env and dependencies

Comment (hamidzr) [reply .pre-commit-config.yaml:78]: also it's a required field:
```
An error has occurred: InvalidConfigError:
==> File .pre-commit-config.yaml
==> At Config()
==> At key: repos
==> At Repository(repo='https://github.com/pycqa/flake8')
=====> Missing required key: rev
```

**Pull request merged**